### DISCUSSION TITLE
RLS Policy table name automatically added to column name

### DISCUSSION THREAD

**Discussion by franciscolourenco** [supabase/supabase#11474] [Questions] [answered]
I'm trying to allow the deletion of projects by members with a role which has permissions to delete projects. What I'm not being able to do is to select only members of the project being deleted.

This is the check expression I wrote:
<img width="1170" alt="image" src="https://user-images.githubusercontent.com/208149/210877639-de8c0adc-d5a0-4e7d-add7-9004a25cd89e.png">

But after saving, the table name get's automatically prefixed to the column name. I believe this causes the where clause to be always true:  
<img width="1166" alt="image" src="https://user-images.githubusercontent.com/208149/210877624-e347cb08-219d-40ba-a2d5-d9286881b44f.png">

I've been able to work around this issue by putting the select in a separate function with a parameter name different than project_id, so that it doesn't get automatically prefixed, however it feels cumbersome to always have to create an extra function for this. 

Is there a way to accomplish the same without an extra function?

```SQL
(auth.uid() IN ( SELECT members.user_id
   FROM (members
     JOIN member_roles ON ((members.role_id = member_roles.id)))
  WHERE ((members.project_id = project_id) AND (member_roles.permission_to_delete_projects = true))))
```

Thanks!

**Comment by GaryAustin1** [comment, marked as answer]
Because you have the same column name it assumes using the table in the FROM.  You should be able to put projects.project_id there to get the column from the table you are doing RLS on.

**Reply by franciscolourenco** [reply]
Ah you are right, my bad, it works fine for permissions in the `projects` table. What about permissions in the `members` table? For example, only project admins should be able to `insert` members. I tried `members.project_id = members.project_id` but members for all projects would be returned? 


```SQL
(auth.uid() IN ( SELECT members.user_id
   FROM (members
     JOIN member_roles ON ((members.role_id = member_roles.id)))
  WHERE ((members.project_id = project_id) AND (member_roles.permission_to_add_members = true))))
```
